model/common/src/icon4py/model/common/math/helpers.py

```diff
@@ -193,6 +193,19 @@ def dot_product_on_cells(
     return x1 * x2 + y1 * y2 + z1 * z2
 
 
+@gtx.field_operator
+def dot_product_on_vertices(
+    x1: fa.VertexField[ta.wpfloat],
+    x2: fa.VertexField[ta.wpfloat],
+    y1: fa.VertexField[ta.wpfloat],
+    y2: fa.VertexField[ta.wpfloat],
+    z1: fa.VertexField[ta.wpfloat],
+    z2: fa.VertexField[ta.wpfloat],
+) -> fa.VertexField[ta.wpfloat]:
+    """Compute dot product of cartesian vectors (x1, y1, z1) * (x2, y2, z2)"""
+    return x1 * x2 + y1 * y2 + z1 * z2
+
+
```

Comment:
I ended up adding this operator and the few next ones while I was using field operators for the RBF computation but ended up removing it in the end. Technically this isn't needed anymore for this PR, but probably belongs here anyway. Keep?

Reply:
Personally I am against adding code which is not needed, just for the case that maybe some day it might be useful.  This is the reason ICON-F90 is so bloated.

Reply:
Removed the added operators.